enhancements/ingress/operator-resource-configuration.md

```diff
@@ -0,0 +1,900 @@
+```
+
+**Option 2: Enable only the specific feature gate using patch command:**
+```bash
+oc patch featuregate cluster --type merge --patch '{"spec":{"featureSet":"CustomNoUpgrade","customNoUpgrade":{"enabled":["IngressRouterResourceLimits"]}}}'
+```
+
+**Option 3: Apply a custom FeatureGate configuration file:**
+```yaml
+apiVersion: config.openshift.io/v1
+kind: FeatureGate
+metadata:
+  name: cluster
+spec:
+  featureSet: CustomNoUpgrade
+  customNoUpgrade:
+    enabled:
+    - IngressRouterResourceLimits
+```
+
+Apply with:
+```bash
+oc apply -f featuregate.yaml
+```
+
+**Note**: Enabling feature gates may require cluster components to restart. For 
+production environments, test in non-production clusters first. Using `TechPreviewNoUpgrade` 
+or `CustomNoUpgrade` means the cluster cannot be upgraded and should only be used for 
+testing.
+
+#### API Changes
+
+Add a new field to the existing `IngressControllerSpec` in the v1 API:
+
+```go
+package v1
+
+import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    corev1 "k8s.io/api/core/v1"
+)
+
+// IngressControllerSpec is the specification of the desired behavior of the IngressController.
+type IngressControllerSpec struct {
+    // ... existing v1 fields ...
+
+    // tuning defines parameters for tuning the performance of ingress controller pods.
+    // +optional
+    Tuning *IngressControllerTuning `json:"tuning,omitempty"`
+
+    // resources defines resource requirements (requests and limits) for the
+    // router pods (HAProxy containers). This field allows setting resource limits
+    // to achieve Guaranteed QoS class for router pods.
+    //
+    // When this field is set, it takes precedence over spec.nodePlacement.resources
+    // for configuring router pod resources.
+    //
+    // When not specified, defaults to:
+    //   router container:
+    //     requests: cpu: 200m, memory: 256Mi
+    //     limits: none (Burstable QoS)
+    //
+    // To achieve Guaranteed QoS, set limits equal to requests:
+    //   resources:
+    //     routerContainer:
+    //       requests:
+    //         cpu: 200m
+    //         memory: 256Mi
+    //       limits:
+    //         cpu: 200m
+    //         memory: 256Mi
+    //
+    // Note: Changing these values will cause router pods to perform a rolling restart.
+    //
+    // +optional
+    // +openshift:enable:FeatureGate=IngressRouterResourceLimits
+    Resources *RouterResourceRequirements `json:"resources,omitempty"`
+}
+
+// RouterResourceRequirements defines resource requirements for ingress router pod containers.
+type RouterResourceRequirements struct {
+    // routerContainer specifies resource requirements (requests and limits) for the
+    // router (HAProxy) container in router pods.
+    //
+    // If not specified, defaults to:
+    //   requests: cpu: 200m, memory: 256Mi
+    //   limits: none
+    //
+    // +optional
+    RouterContainer *corev1.ResourceRequirements `json:"routerContainer,omitempty"`
```

Comment:
ResourceRequirements also has a Claims subfield: https://pkg.go.dev/k8s.io/api/core/v1#ResourceRequirements , would we plan on allowing the user to set that?